powersimdata/design/investment/investment_costs.py

```diff
@@ -67,7 +67,8 @@ def calculate_ac_inv_costs(
     """
     grid = scenario.state.get_grid()
     if base_grid is None:
-        base_grid = Grid(
+        base_grid = Grid( # Why are we repeatedly regenerating the base_grid?
+                          # (There are 3 other instances of this in this file)
```

Comment:
This repeated initialization (3 times in this file) of the `Grid` is pretty rough. What do you all think about adding `base_grid` to the scenario class? It seems to make sense that the user would always want the `base_grid` associated with the scenario they're creating

Reply:
I guess it would not hurt to have it as an attribute of the `Grid` class. What do you thing @BainanXia and @danielolsen?

Reply:
The point of the Grid caching is so that the repeated initialization doesn't cost much extra time. Initializing another grid object every time we load a Scenario seems like it would hurt performance, not help it.

Reply:
I imagine we are talking about adding this to the scenario class:
```
def get_base_grid(self):
    return Grid(self.info["interconnect"].split("_"), source=self.info["grid_model"])
```

where the result would be cached internally. However if the `Grid` constructor takes in `data` instead, we'd have to move the cache elsewhere, maybe a "factory" that essentially does the if/else logic currently in the grid class.

Reply:
There are two things we are discussing here:
- Having a grid attribute `self.grid = Grid(self.info["interconnect"].split("_"), source=self.info["grid_model"])` in the scenario class, so that we could call `scenario.grid` instead of `scenario.get_grid()`.
- Having a base grid attribute or a method that returns the base grid in scenario class.

For the first one, we could evaluate whether there are situations that we load a scenario object without querying `get_grid()` method. However, it might add more overheads when load a scenario, which is probably okay I presume.

For the second one, I think it is obvious that we don't need the base grid for many situations when loading a scenario object. I would not set the base grid as an attribute and I'm with @danielolsen at this point, that's the point of caching.